### DISCUSSION TITLE
Creating a Dynamic Path based on the date the project was initiated

### DISCUSSION THREAD

**Discussion by kaelabruin** [bonsai-rx/bonsai#2013] [Q&A]
Hi everyone, 

I want to have bonsai help me organize my files, such that everytime a bonsai workflow is ran it saves all the files to a folder with the Date (MM-DD-YYYY) and also time (HH-MM-SS).  For example I would want the path to be like {data_folder}\{project_name}\{camera_name}\{session_timestamp}\00.mp4. 

I am going to be using the same workflow, but automatically stopping and restarting it at the end of each day.
So far I have been trying to use the externalize filename function, but this only seems to change the filename and not the path itself. 

Is there any way to directly change the path within bonsai?

I attached an image of my workflow so far below. 

![image](https://github.com/user-attachments/assets/cbfc3c2e-a463-424e-8c4d-a0db58cfd79d)

Thanks!

Kai

**Comment by bruno-f-cruz** [comment]
Please also include the workflow file.

In general (and I am not sure I fully understood the question) you can make directories by simply using the OS path separator \ or /. Eg:

Format(`{0}\{1}`)

**Reply by kaelabruin** [reply]
Thank you! I seem to be unable to upload the workflow due to the unsupported file type.

Will this directory change the date/path for whenever I run the bonsai workflow on a different day? I want the path to include a folder that is named based on the date. I tried doing data_folder/project_name/camera_name/{0}/%02d.csv, but this path did not change the {0} to be the current date.

**Reply by bruno-f-cruz** [reply]
Please attach the workflow as a zip file.

**Reply by kaelabruin** [reply]
Got it!

Also do you know how I could use the property mapping node for this goal?

[autostart_dates_01.n_cams=1.cams=1.crf=28.fps=50.v04.zip](https://github.com/user-attachments/files/16971947/autostart_dates_01.n_cams.1.cams.1.crf.28.fps.50.v04.zip)

**Reply by kaelabruin** [reply]
[autostart_dates_01.n_cams=1.cams=1.crf=28.fps=50.v04 (2).zip](https://github.com/user-attachments/files/16984169/autostart_dates_01.n_cams.1.cams.1.crf.28.fps.50.v04.2.zip)
I edited the workslow to get closer to my goal, I just need help to be able to adjust the date for both the video file and the csv file now.

**Reply by kaelabruin** [reply]
[autostart_dates_01.n_cams=1.cams=1.crf=28.fps=50.v06.zip](https://github.com/user-attachments/files/16985272/autostart_dates_01.n_cams.1.cams.1.crf.28.fps.50.v06.zip)
I feel like I am close to the solution now, but I am getting a "pipe is broken" error whenever I try to run the workflow.

**Reply by yqwang1** [reply]
> [autostart_dates_01.n_cams=1.cams=1.crf=28.fps=50.v06.zip](https://github.com/user-attachments/files/16985272/autostart_dates_01.n_cams.1.cams.1.crf.28.fps.50.v06.zip) I feel like I am close to the solution now, but I am getting a "pipe is broken" error whenever I try to run the workflow.

Hi Just curious did you get a solution? I also got the same error
